src/main/java/io/cdap/http/internal/HttpResourceModel.java

```diff
@@ -195,6 +204,17 @@ private Object getHeaderParamValue(Map<Class<? extends Annotation>, ParameterInf
     return hasHeader ? info.convert(request.headers().getAll(headerName)) : info.convert(defaultValue(annotations));
   }
 
+  @SuppressWarnings("unchecked")
+  private Object getCookieParamValue(Map<Class<? extends Annotation>, ParameterInfo<?>> annotations,
+                                     HttpRequest request, Map<String, Cookie> cookies) throws Exception {
+    ParameterInfo<Cookie> info = (ParameterInfo<Cookie>) annotations.get(CookieParam.class);
+    CookieParam cookieParam = info.getAnnotation();
+    String cookieName = cookieParam.value();
+    boolean hasCookie = cookies.containsKey(cookieName);
+//    return hasCookie ? info.convert(cookies.get(cookieName)) : info.convert(defaultValue(annotations));
```

Comment:
Please remove commented code

Reply:
Also, why the commented line doesn't work? Generally we don't want to have `null` if default value is provided.

Reply:
Sorry, didn't realise this pertained to the `@DefaultValue` annotations. I've added support for this as well. Thank you for a prompt review!